src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php

```diff
@@ -0,0 +1,292 @@
+            return $this->applyNoMultiLineFix($tokens);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     */
+    protected function applyNoMultiLineFix(Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        foreach ($tokens as $index => $token) {
+            if (!$token->equals(';')) {
+                continue;
+            }
+
+            $previousIndex = $index - 1;
+            $previous = $tokens[$previousIndex];
+            if (!$previous->isWhitespace() || false === strpos($previous->getContent(), "\n")) {
+                continue;
+            }
+
+            $content = $previous->getContent();
+            if (("\n" === $content[0] || "\r" === $content[0]) && $tokens[$index - 2]->isComment()) {
+                $tokens[$previousIndex] = new Token([$previous->getId(), $lineEnding]);
+            } else {
+                $tokens->clearAt($previousIndex);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     */
+    protected function applyChainedCallsFix(Tokens $tokens)
+    {
+        for ($index = count($tokens) - 1; $index >= 0; --$index) {
+            // continue if token is not a semicolon
+            if (!$tokens[$index]->equals(';')) {
+                continue;
+            }
+
+            // get the indent of the chained call, null in case it's not a chained call
+            $indent = $this->getIndentation($index - 1, $tokens);
+
+            if (null === $indent) {
+                continue;
+            }
+
+            // unset semicolon
+            $tokens->clearAt($index);
+
+            // find the line ending token index after the semicolon
+            $index = $this->getNewLineIndex($index, $tokens);
+
+            // line ending string of the last method call
+            $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+            // appended new line to the last method call
+            $newline = new Token([T_WHITESPACE, $lineEnding.$indent]);
+
+            // insert the new line with indented semicolon
+            $tokens->insertAt($index, [$newline, new Token(';')]);
+        }
+    }
+
+    /**
+     * Find the index for the new line. Return the given index when there's no new line.
+     *
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return int
+     */
+    private function getNewLineIndex($index, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index, $count = count($tokens); $index < $count; ++$index) {
+            if (false !== strstr($tokens[$index]->getContent(), $lineEnding)) {
+                return $index;
+            }
+        }
+
+        return $index;
+    }
+
+    /**
+     * Find the indentation of the chained call at $index.
+     *
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return null|string
+     */
+    private function getIndentation($index, Tokens $tokens)
```

Comment:
What does this method actually do?

Finds this part underlined part `___`?

```php
$this->methodCall()
____->anotherCall();
```

If so, finding first previous `T_WHITESPACE` token with line break would be easier and more clear

Reply:
Thank you for reviewing @TomasVotruba! I've renamed the method and gave an example, hope this makes it more clear :)